pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionarySingleColumnGroupKeyGenerator.java

```diff
@@ -72,47 +78,122 @@ public int getGlobalGroupKeyUpperBound() {
   @Override
   public void generateKeysForBlock(TransformBlock transformBlock, int[] groupKeys) {
     BlockValSet blockValSet = transformBlock.getBlockValueSet(_groupByExpression);
+    RoaringBitmap nullBitmap = blockValSet.getNullBitmap();
     int numDocs = transformBlock.getNumDocs();
 
     switch (_storedType) {
       case INT:
         int[] intValues = blockValSet.getIntValuesSV();
+        if (_nullHandlingEnabled) {
+          if (nullBitmap != null && nullBitmap.getCardinality() < numDocs) {
+            for (int i = 0; i < numDocs; i++) {
+              groupKeys[i] = nullBitmap.contains(i) ? getKeyForNullValue() : getKeyForValue(intValues[i]);
+            }
+          } else if (numDocs > 0) {
+            _groupIdForNullValue = _numGroups++;
+            Arrays.fill(groupKeys, _groupIdForNullValue);
```

Comment:
There might already be an id associated with `null`
```suggestion
            Arrays.fill(groupKeys, getKeyForNullValue());
```

Reply:
Not sure I follow. I will make the change but would like to understand under what circumstances there might be already an id associated w/ `null`? This code gets executed when input is all null values.

Reply:
This method is called per block instead of only once. In the previous block, the null key might already be assigned